animation_nodes/nodes/spline/offset_spline.py

```diff
@@ -0,0 +1,83 @@
+    bl_label = "Offset Spline"
+    errorHandlingType = "EXCEPTION"
+
+    def checkedPropertiesChanged(self, context):
+        self.updateSocketVisibility()
+        executionCodeChanged()
+
+    useLocation: BoolProperty(update = checkedPropertiesChanged)
+    useRadius: BoolProperty(update = checkedPropertiesChanged)
+    useTilt: BoolProperty(update = checkedPropertiesChanged)
+
+    useVectorList: VectorizedSocket.newProperty()
+    useRadiusList: VectorizedSocket.newProperty()
+    useTiltList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Spline", "Stroke", "spline", dataIsModified = True)
+        self.newInput("Falloff", "Falloff", "falloff")
+        self.newInput(VectorizedSocket("Vector", "useVectorList",
+            ("Location", "offsetLocations", dict(value = (0, 0, 1))),
+            ("Locations", "offsetLocations")))
+        self.newInput(VectorizedSocket("Float", "useRadiusList",
+            ("Radius", "offsetRadius", dict(value = 1)),
+            ("Radii", "offsetRadii")))
+        self.newInput(VectorizedSocket("Float", "useTiltList",
+            ("Tilt", "offsetTilt", dict(value = 1)),
+            ("Tilts", "offsetTilts")))
+
+        self.newOutput("Spline", "Stroke", "spline")
+
+        self.updateSocketVisibility()
+
+    def draw(self, layout):
+        col = layout.column(align = True)
+
+        row = col.row(align = True)
+        row.prop(self, "useLocation", text = "Location", toggle = True)
+
+        row = col.row(align = True)
+        row.prop(self, "useRadius", text = "Radius", toggle = True)
+        row.prop(self, "useTilt", text = "Tilt", toggle = True)
+
+    def updateSocketVisibility(self):
+        self.inputs[2].hide = not self.useLocation
+        self.inputs[3].hide = not self.useRadius
+        self.inputs[4].hide = not self.useTilt
+
+    def execute(self, spline, falloff, offsetLocations, offsetRadii, offsetTilts):
+        if not any((self.useLocation, self.useRadius, self.useTilt)):
+            return spline
+
+        falloffEvaluator = self.getFalloffEvaluator(falloff)
+
+        if self.useRadius:
+            _offsetRadii = VirtualDoubleList.create(offsetRadii, 0)
+            offsetFloatList(spline.points, spline.radii, _offsetRadii, falloffEvaluator)
+
+        if self.useTilt:
+            _offsetTilts = VirtualDoubleList.create(offsetTilts, 0)
+            offsetFloatList(spline.points, spline.tilts, _offsetTilts, falloffEvaluator)
+
+        if self.useLocation:
+            _offsetLocations = VirtualVector3DList.create(offsetLocations, (0, 0, 0))
+            influences = VirtualFloatList.create(falloffEvaluator.evaluateList(spline.points), 0)
+            offsetVector3DList(spline.points, _offsetLocations, influences)
+            if spline.type == "BEZIER":
+                offsetVector3DList(spline.leftHandles, _offsetLocations, influences)
+                offsetVector3DList(spline.rightHandles, _offsetLocations, influences)
```

Comment:
You actually implemented something really nice here. You only evaluated the falloff a single time. But we can take this one step further and use the influences for the rest of the offsets as well. In which case, `offsetFloatList` would need to be rewritten to be similar to `offsetVector3DList`.

However, note that the influences needn't be a virtual list. And the `invert` parameter is redundant in this case as well, so it can be removed.

Also, in this case, it wound't make sense to put the functions in an `falloffs.pyx` file. I would say the structure should be:

`offsetFloats` -> `/nodes/number/c_utils.pyx`
`offset3DVectors` -> `/nodes/vector/c_utils.pyx`

You may update the *Offset GP Stroke* as well if you want.

Reply:
> In which case, offsetFloatList would need to be rewritten to be similar to offsetVector3DList.
> You may update the Offset GP Stroke as well if you want.

I was just going to ask for this.

Omar, should we change the name of `offsetFloats, offset3DVectors` these functions something else? Because it may confuse with the `offset3DVectors` function from `vector/offset_vector.pyx`.

Reply:
The function in `offset_vector.pyx` is called `offsetVector3DList` (Or are you talking about something else?), that's why I was proposing those function names:

`offsetFloats()` -> /nodes/number/c_utils.pyx
`offset3DVectors()` -> /nodes/vector/c_utils.pyx

Reply:
Ohh sorry, I did not notice that.

Reply:
Done.